test/test-pkg-config.rb

```diff
@@ -314,4 +314,34 @@ def test_equals_to
                    parse_requires("fribidi = 1.0"))
     end
   end
+
+  sub_test_case("#merge_back_cflags") do
+    def merge_back_cflags(cflags)
+      @glib.__send__(:merge_back_cflags, cflags)
+    end
+
+    def test_d_flag
+      assert_equal(["-DFOO"],
+                   merge_back_cflags(["-DFOO", "-DFOO"]))
+    end
+
+    def test_w_flag
+      assert_equal(["-Wno-unknown-warning-option"],
+                   merge_back_cflags(["-Wno-unknown-warning-option",
+                                      "-Wno-unknown-warning-option"]))
+    end
+
+    def test_wa_wl_wp_flags_not_merged_back
```

Comment:
We don't need "expected" information in test name. It's redundant. Test code should show it.

```suggestion
    def test_wa_wl_wp
```